core/src/main/java/org/elasticsearch/index/query/MissingQueryBuilder.java

```diff
@@ -30,18 +40,29 @@
 
     public static final String NAME = "missing";
 
-    private String name;
+    public static final boolean DEFAULT_NULL_VALUE = false;
 
-    private Boolean nullValue;
+    public static final boolean DEFAULT_EXISTENCE_VALUE = true;
 
-    private Boolean existence;
+    private final String fieldPattern;
+
+    private Boolean nullValue = DEFAULT_NULL_VALUE;
+
+    private Boolean existence = DEFAULT_EXISTENCE_VALUE;
 
     static final MissingQueryBuilder PROTOTYPE = new MissingQueryBuilder(null);
 
-    public MissingQueryBuilder(String name) {
-        this.name = name;
+    public MissingQueryBuilder(String fieldPattern) {
+        if (fieldPattern == null) {
+            throw new IllegalArgumentException("missing must be provided with a field");
```

Comment:
Does this work? Above PROTOTYPE is initialized with null which will result in an exception being thrown here.